prow/apis/prowjobs/v1/types.go

```diff
@@ -532,7 +532,22 @@ func (g *GCSConfiguration) Validate() error {
 
 // ProwJobStatus provides runtime metadata, such as when it finished, whether it is running, etc.
 type ProwJobStatus struct {
-	StartTime      metav1.Time  `json:"startTime,omitempty"`
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// PreviousPodCreateTime holds the creation time of the pod which was previously
+	// created for the job and unexpectedly disappeared
+	PreviousPodCreateTime *metav1.Time `json:"PreviousPodCreateTime,omitempty"`
```

Comment:
I don't think we can create properties that reference pods, because depending on the `agent` there may be no pod. Maybe we can instead use a `[]Condition` that has a free-text `name` field and a `duration` field?

Reply:
I can see why this would not work for a pipeline, but for instance plank also stores the pod name itself also in the spec, so just checking the agent might be good enough?

Another option would be to let directly plank create the metrics. Then they would not be needed on the spec, and plank has access to the pod.